src/api/app/controllers/webui/users/tokens/users_controller.rb

```diff
@@ -0,0 +1,34 @@
+class Webui::Users::Tokens::UsersController < Webui::WebuiController
+  before_action :set_token
+
+  def index
+    @users = @token.users_shared_among
+    @groups = @token.groups_shared_among
+  end
+
+  def create
+    user = User.find_by(login: params[:userid])
+
+    if @token.users_shared_among.include?(user)
+      redirect_back(fallback_location: root_path, notice: "User #{user.login} is already associated to the token")
+    elsif @token.users_shared_among << user
+      redirect_back(fallback_location: root_path, success: "User #{user.login} has been added to the token successfully")
+    end
```

Comment:
Same here basically. `if @token.users_shared_among.include?(user)` you are done. No need to error.

Reply:
This is not an error but a 'notice', just an informative blue flash message.

Reply:
Do you still prefer to do nothing?

Reply:
See above.